sos/plugins/podman.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (C) 2018 Red Hat, Inc. Daniel Walsh <dwalsh@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin, UbuntuPlugin
+
+
+class Podman(Plugin):
+
+    """Podman containers
+    """
+
+    plugin_name = 'podman'
+    profiles = ('container',)
+
+    option_list = [
+        ("all", "enable capture for all containers, even containers "
+            "that have terminated", 'fast', False),
+        ("logs", "capture logs for running containers",
+            'fast', False),
+        ("size", "capture image sizes for podman ps", 'slow', False)
+    ]
+
+    def setup(self):
+        self.add_copy_spec([
+            "/etc/containers/registries.conf",
+            "/etc/containers/storage.conf",
+            "/etc/containers/mounts.conf",
+            "/etc/containers/policy.json",
+        ])
+
+        subcmds = [
+            'info',
+            'images',
+            'pod ps',
+            'pod ps -a',
+            'ps',
+            'ps -a',
+            'stats --no-stream',
+            'version',
+        ]
+
+        for subcmd in subcmds:
+            self.add_cmd_output("podman %s" % subcmd)
```

Comment:
We can write that a little more compactly as:

```python
    self.add_cmd_output(["podman %s" % s for s in subcmds])
```

That said, this pattern is appearing in four plugins now (plus this PR). I'm inclined to think we should add a `add_cmd_with_subcmds()` method (or a less horrible name):

```python
    subcmds = [...]
    self.add_cmd_with_subcmds("podman %s", subcmds)
```

We're discussing a change at the moment to eliminate more explicit iteration and conditions in plugins (the eventual aim is that most will be purely declarative - just member variables and little or no executable code), and this kind of interface would fit into that.

Reply:
Thanks - I'll think about the subcommand interface some more & talk to the rest of the folks about it: we'll update the plugin to use it once it lands.